source/jormungandr/tests/ptref_tests.py

```diff
@@ -778,6 +778,121 @@ def test_pt_objects_with_disable_disruption(self):
         )
         assert len(response['disruptions']) == 1
 
+    def test_networks_with_external_code(self):
+        """test networks with external_code parameter"""
+        response = self.query_region("networks")
+        networks = get_not_null(response, 'networks')
+        assert len(networks) == 4
+        assert networks[0]['id'] == 'network:A'
+        assert networks[1]['id'] == 'network:B'
+        assert networks[2]['id'] == 'network:CaRoule'
+        assert networks[3]['id'] == 'network:freq'
+
+        response = self.query_region("networks?external_code=A")
+        networks = get_not_null(response, 'networks')
+        assert len(networks) == 1
+        assert networks[0]['id'] == 'network:A'
+
+        response = self.query_region("networks?external_code=B")
+        networks = get_not_null(response, 'networks')
+        assert len(networks) == 1
+        assert networks[0]['id'] == 'network:B'
+
+        response = self.query_region("networks?external_code=wrong_code")
+        networks = get_not_null(response, 'networks')
+        assert len(networks) == 4
+
+        # without coverage
+        response, code = self.query_no_assert("v1/networks?external_code=A")
+        assert code == 200
+        networks = get_not_null(response, 'networks')
+        assert len(networks) == 1
+        assert networks[0]['id'] == 'network:A'
+
+        response, code = self.query_no_assert("v1/networks?external_code=wrong_code")
+        assert code == 404
+        message = get_not_null(response, 'message')
+        assert 'Unable to find an object for the uri wrong_code' in message
```

Comment:
Why the difference of behavior? 
- With a coverage, "wrong_code" returns all networks
- Without a coverage, "wrong_code" returns an error

Reply:
I think you've right. it should be the same behavior. Fix it fix it :eagle: